### ISSUE TITLE
Group unique name column not taken into account

### ISSUE THREAD

**Issue by vilius-rugaitis** [sonata-project/SonataUserBundle#794]
### Environment

Mac OSX + XAMPP 5.6.24-1
#### Sonata packages

``` bash
$ composer show sonata-project/*
sonata-project/admin-bundle              3.8.0  The missing Symfony Admin Generator
sonata-project/block-bundle              3.2.0  Symfony SonataBlockBundle
sonata-project/cache                     1.0.7  Cache library
sonata-project/core-bundle               3.1.1  Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.2    Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.0.2  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.1.0  Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.1.10 Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.7.0  Lightweight Exporter library
sonata-project/google-authenticator      1.0.2  Library to integrate Google Authenticator into a PHP project
sonata-project/user-bundle               3.0.1  Symfony SonataUserBundle
```
#### Symfony packages

``` bash
$ composer show symfony/*
symfony/monolog-bundle     2.11.1  Symfony MonologBundle
symfony/phpunit-bridge     v2.8.11 Symfony PHPUnit Bridge
symfony/polyfill-apcu      v1.2.0  Symfony polyfill backporting apcu_* functions to lower PHP versions
symfony/polyfill-intl-icu  v1.2.0  Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring  v1.2.0  Symfony polyfill for the Mbstring extension
symfony/polyfill-php54     v1.2.0  Symfony polyfill backporting some PHP 5.4+ features to lower PHP versions
symfony/polyfill-php55     v1.2.0  Symfony polyfill backporting some PHP 5.5+ features to lower PHP versions
symfony/polyfill-php56     v1.2.0  Symfony polyfill backporting some PHP 5.6+ features to lower PHP versions
symfony/polyfill-php70     v1.2.0  Symfony polyfill backporting some PHP 7.0+ features to lower PHP versions
symfony/polyfill-util      v1.2.0  Symfony utilities for portability of PHP codes
symfony/security-acl       v3.0.0  Symfony Security Component - ACL (Access Control List)
symfony/swiftmailer-bundle v2.3.11 Symfony SwiftmailerBundle
symfony/symfony            v2.8.11 The Symfony PHP framework
```
#### PHP version

``` bash
$ php -v
PHP 5.5.36 (cli) (built: May 29 2016 01:07:06) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2015 Zend Technologies
```
## Subject

I have experienced a problem, where creating a group with already existing name leads to an exception. Why is this error not handled directly over the doctrine restrictions? Please see attached screenshot.
## Steps to reproduce
- Create a group with name "test"
- Try to create a second group with name "test"
## Expected results
- Error message with "Group name already exists"
## Actual results

<img width="1006" alt="bildschirmfoto 2016-10-07 um 10 35 02" src="https://cloud.githubusercontent.com/assets/22676831/19183918/dde82674-8c7a-11e6-8338-2792700fa750.png">

**Comment by core23** [comment]
Not sure if it is a bug on our side. There is no validation for unique group names in the `FOSUserBundle` too.

**Comment by vilius-rugaitis** [comment]
So, should I address the issue in the `FOSUserBundle` repo?

**Comment by core23** [comment]
Correct

**Comment by OskarStark** [comment]
Closed via #869
